src/test/scala/circtTests/stage/ChiselStageSpec.scala

```diff
@@ -1231,6 +1245,17 @@ class ChiselStageSpec extends AnyFunSpec with Matchers with chiselTests.Utils {
       (ChiselStage.emitCHIRRTL(new RawModule {}) should not).include("LogLevelAnnotation")
       (ChiselStage.emitCHIRRTL(new RawModule {}, Array("-ll", "info")) should not).include("LogLevelAnnotation")
     }
+
+    it("should allow remapping one layer to another") {
+      val chirrtl = ChiselStage.emitCHIRRTL(
+        new ChiselStageSpec.LayerRemappingTest,
+        Array("--remap-layer", "circtTests.stage.ChiselStageSpec$A$,circtTests.stage.ChiselStageSpec$B$")
```

Comment:
IMO, it would be better if we could make this work using the expected Scala package path of the object rather than the actual classname, ie. I would prefer to see this as:
```scala
Array("--remap-layer", "circtTests.stage.ChiselStageSpec.A,circtTests.stage.ChiselStageSpec.B")
```
We already require layers to be Singletons so there's no risk of ambiguity, we can insert and remove the `$` ourselves and just include them in error messages as appropriate (e.g. we could print the thing we couldn't find in case something is going wrong).

WDYT?

Reply:
I agree. However, `.` is annoyingly ambiguous as any number of these could definitely be package or object separators (and could also be a literal `.`, maybe):

1. `.` 
2. `$`
3. `$u002E` (In the weird case of `` object `A.B` ``.)

I think an algorithm could work that tries all possibilities of, for the provided example:

1. `circtTests.stage.ChiselStageSpec.A` (Maybe the user means the class and made a mistake?)
1. `circtTests.stage.ChiselStageSpec.A$`
2. `circtTests.stage.ChiselStageSpec$A$`
3. `circtTests.stage$ChiselStageSpec$A$`
4. `circtTests$stage$ChiselStageSpec$A$`

However, it seemed weird to do this "guess and check" approach and instead require the user to just specify the exact object that they want. Given that this is intended to be a very advanced user API, I figured the precision was appreciated.

WDYT? Do you want/like the guess and check algorithm?

Reply:
Fair points, and we can also just go with this for now and add something more sugary later.

Also maybe Scala reflection gives something nicer here where the name munging is hidden under a more intentional API...

In any case, I think we should move forward with this and do that later.

Reply:
Sounds good. I'll keep this as-is and we can revisit it later. I'm also very unconcerned about changing this advanced API in the future if we need to. Thanks for the discussion.

Reply:
Agreed on all counts! Thanks!